beacon_node/beacon_chain/src/validator_monitor.rs

```diff
@@ -338,93 +345,123 @@ impl<T: EthSpec> ValidatorMonitor<T> {
+                } else {
+                    metrics::inc_counter_vec(
+                        &metrics::VALIDATOR_MONITOR_PREV_EPOCH_ON_CHAIN_ATTESTER_MISS,
+                        &[id],
+                    );
+                    error!(
+                        self.log,
+                        "Previous epoch attestation missing";
+                        "epoch" => prev_epoch,
+                        "validator" => id,
+                    )
+                }
 
-                    if summary.is_previous_epoch_attester {
-                        metrics::inc_counter_vec(
-                            &metrics::VALIDATOR_MONITOR_PREV_EPOCH_ON_CHAIN_ATTESTER_HIT,
-                            &[id],
-                        );
-                    } else {
-                        metrics::inc_counter_vec(
-                            &metrics::VALIDATOR_MONITOR_PREV_EPOCH_ON_CHAIN_ATTESTER_MISS,
-                            &[id],
-                        );
-                    }
-                    if summary.is_previous_epoch_head_attester {
-                        metrics::inc_counter_vec(
-                            &metrics::VALIDATOR_MONITOR_PREV_EPOCH_ON_CHAIN_HEAD_ATTESTER_HIT,
-                            &[id],
-                        );
-                    } else {
-                        metrics::inc_counter_vec(
-                            &metrics::VALIDATOR_MONITOR_PREV_EPOCH_ON_CHAIN_HEAD_ATTESTER_MISS,
-                            &[id],
-                        );
-                        warn!(
-                            self.log,
-                            "Attested to an incorrect head";
-                            "epoch" => prev_epoch,
-                            "validator" => id,
-                        );
-                    }
-                    if summary.is_previous_epoch_target_attester {
-                        metrics::inc_counter_vec(
-                            &metrics::VALIDATOR_MONITOR_PREV_EPOCH_ON_CHAIN_TARGET_ATTESTER_HIT,
-                            &[id],
-                        );
-                    } else {
-                        metrics::inc_counter_vec(
-                            &metrics::VALIDATOR_MONITOR_PREV_EPOCH_ON_CHAIN_TARGET_ATTESTER_MISS,
-                            &[id],
-                        );
+                // Indicates if any on-chain attestation hit the head.
+                if previous_epoch_matched_head {
+                    metrics::inc_counter_vec(
+                        &metrics::VALIDATOR_MONITOR_PREV_EPOCH_ON_CHAIN_HEAD_ATTESTER_HIT,
+                        &[id],
+                    );
+                } else {
+                    metrics::inc_counter_vec(
+                        &metrics::VALIDATOR_MONITOR_PREV_EPOCH_ON_CHAIN_HEAD_ATTESTER_MISS,
+                        &[id],
+                    );
+                    warn!(
+                        self.log,
+                        "Attested failed to match head";
```

Comment:
The wording is strange here

```suggestion
                        "Attestation failed to match head";
```